src/terrain/generation/land_generator.cpp

```diff
@@ -29,13 +29,20 @@ namespace terrain {
 
 namespace generation {
 
-LandGenerator::LandGenerator(
-    const std::map<MaterialId, const Material>& materials_, const Json::Value data
-) :
-    current_region(0),
-    current_sub_region(0), materials(materials_), data_(data) {}
-
-// LandGenerator::LandGenerator() : current_region(0), current_sub_region(0) {}
+LandGenerator::LandGenerator(const Json::Value data) :
+    current_region(0), current_sub_region(0) {
+    for (const Json::Value& region : data) {
+        std::shared_ptr<JsonToTile> stamp_generator;
+        if (region["type"] == "Positions") {
+            stamp_generator = std::make_shared<FromPosition>(FromPosition(region));
+        } else if (region["type"] == "Radius") {
+            stamp_generator = std::make_shared<FromRadius>(FromRadius(region));
+        } else if (region["type"] == "Grid") {
+            stamp_generator = std::make_shared<FromGrid>(FromGrid(region));
+        }
+        stamp_generators_.push_back(stamp_generator);
+    }
+}
```

Comment:
```suggestion
    for (const Json::Value& region : data) {
        std::shared_ptr<JsonToTile> stamp_generator{};
        
        switch(region["type"][0]) {
            case 'P': // Positions
                assert(region["type"] == "Positions");
                stamp_generator = std::make_shared<FromPosition>(region);
                break;
                
            case 'R': // Radius
                assert(region["type"] == "Radius");
                stamp_generator = std::make_shared<FromRadius>(region);
                break;
                
            case 'G': // Grid
                assert(region["type"] == "Grid");
                stamp_generator = std::make_shared<FromGrid>(region);
                break;
            
            default:
                LOG_CRITICAL(logger, "Unknown region type {}", region["type");
                abort();
        }
        
        stamp_generators_.push_back(stamp_generator);
    }
}
```

Did you notice you used `std::make_shared` wrong? You were calling the copy ctor.

Reply:
I don't really want a switch statement

Reply:
Its much faster than calling `strcmp`